galasa-ui/src/components/test-runs/TestRunsTabs.tsx

```diff
@@ -27,106 +27,123 @@ interface TestRunsTabProps {
-  ]);
 
-  const [columnsOrder, setColumnsOrder] = useState<{ id: string; columnName: string }[]>(RESULTS_TABLE_COLUMNS);
-  const [isInitialized, setIsInitialized] = useState(false);
+  const TABS_IDS = ['timeframe', 'table-design', 'search-criteria', 'results'];
 
-  // Load from URL parameters first (only on mount)
-  useEffect(() => {
-    const visibleColumnsParam = searchParams.get("visibleColumns");
-    const columnsOrderParam = searchParams.get("columnsOrder");
+  // Initialize selectedIndex based on URL parameters or default to first tab
+  const [selectedIndex, setSelectedIndex] = useState(() => {
+    const tabParam = searchParams.get("tab");
+    const initialIndex = tabParam ? TABS_IDS.indexOf(tabParam) : -1;
+    return initialIndex !== -1 ? initialIndex : 0;
+  });
 
-    if (visibleColumnsParam) {
-      const visibleColumns = visibleColumnsParam.split(",");
-      setSelectedVisibleColumns(visibleColumns);
-    } 
+  // Initialize selectedVisibleColumns  based on URL parameters or default values
+  const [selectedVisibleColumns, setSelectedVisibleColumns] = useState<string[]>(
+    () => searchParams.get('visibleColumns')?.split(',') || [
+      COLUMNS_IDS.SUBMITTED_AT,
+      COLUMNS_IDS.TEST_RUN_NAME,
+      COLUMNS_IDS.REQUESTOR,
+      COLUMNS_IDS.TEST_NAME,
+      COLUMNS_IDS.STATUS,
+      COLUMNS_IDS.RESULT,
+    ]
+  );
 
-    if (columnsOrderParam) {
-      const columnsOrder = columnsOrderParam.split(",").map(id => {
-        const column = RESULTS_TABLE_COLUMNS.find(col => col.id === id);
-        return column ? { id: column.id, columnName: column.columnName } : null;
-      }).filter(Boolean) as { id: string; columnName: string }[];
+  // Initialize columnsOrder based on URL parameters or default to RESULTS_TABLE_COLUMNS
+  const [columnsOrder, setColumnsOrder] = useState<{ id: string; columnName: string }[]>(() => {
+    const orderParam = searchParams.get('columnsOrder');
+    let correctOrder = RESULTS_TABLE_COLUMNS;
 
-      if (columnsOrder.length > 0) {
-        setColumnsOrder(columnsOrder);
-      }
+    // Parse the order from the URL parameter
+    if (orderParam) {
+      correctOrder = orderParam.split(',')
+        .map(id => RESULTS_TABLE_COLUMNS.find(col => col.id === id))
+        .filter(Boolean) as { id: string; columnName: string }[];
     }
 
-    // Mark as initialized after loading
-    setIsInitialized(true);
-  }, []); 
+    return correctOrder;
+  });
+    
+  // State to track if the component has been initialized
+  const [isInitialized, setIsInitialized] = useState(false);
+  useEffect(() => {setIsInitialized(true);}, []);
+
+  // Define the tabs with their corresponding labels
+  const TABS_CONFIG: TabConfig[] = [
+    { id: TABS_IDS[0], label: translations('tabs.timeframe') },
+    { id: TABS_IDS[1], label: translations('tabs.tableDesign') },
+    { id: TABS_IDS[2], label: translations('tabs.searchCriteria') },
+    { id: TABS_IDS[3], label: translations('tabs.results') },
+  ];
```

Comment:
If we used a map for the tabs, then I don't think we would need this list either